src/en/charms-deploying.md

```diff
@@ -1,67 +1,88 @@
 
-In most cases, you will want to deploy charms by fetching them directly from the
-Charm Store. This ensures that you get the relevant, up to date version of the
-charm and "everything just works". To deploy a charm like this you can just
-specify:
+Typically, services are deployed using the online charms. This ensures that
+you get the latest version of the charm. To deploy in this way:
 
 ```bash
 juju deploy mysql
 ```
 
-Running this will do exactly what you expect - fetch the latest Juju charm for
-the series you are running and then use the bootstrap environment to initiate a
-new instance and deploy MySQL
+This will create a machine and use the latest online MySQL charm (for your
+default series) to deploy a MySQL service.
+
+!!! Note: The default series can be configured at a model level (see
+[Configuring model](./models-config.html)). In the absence of this setting,
+the default is to use the Ubuntu version running on the Juju client (i.e. where
+the Juju commands are being invoked).
 
-Juju usefully supports a system of namespaces that means you can actually deploy
-charms from a variety of sources. The default source is the charm store. The
-above command is the same as running:
+Assuming that the Xenial series charm exists and was used above, an equivalent
+command is:
 
 ```bash
-juju deploy cs:precise/mysql
+juju deploy cs:xenial/mysql
 ```
 
-which follows the format:
+Where 'cs' denotes the charm store.
 
-```nohighlight
-<repository>:<series>/<service>
-```
+### Channel support	
 
+The charm store offers charms in different stages of development. Such stages
+are called *channels*.
 
-# Deploying from a local repository
+```bash
+juju deploy mysql --channel channel_name
+```
 
-There are many cases when you may wish to deploy charms from a local filesytem
-source rather than the charm store:
+Such a channel will be used if the charm's revision is:
 
-- When testing charms you have written.
-- When you have modified store charms for some reason.
-- When you don't have direct internet access.
+ 1. undefined
+ 1. defined only for the specified non-stable channel
 
-... and probably a lot more times which you can imagine yourselves.
+Each channel will have a "pointer" that redirects to a certain *revision*.
 
-You may substitute a path pointing to a directory to deploy a
-'local' charm:
+#### Charm upgrades
+Because the pointer can fluctuate among revisions it is possible that during a
+charm upgrade the channel revision is different than the revision of a
+currently deployed charm. The following rules apply:
 
+- If a channel revision is older, downgrade the deployed charm to that revision
+- If a channel revision is newer, upgrade the deployed charm to that revision
+
+Below we specify a channel with the `charm-upgrade` command:
 
 ```bash
-juju deploy ./charms/wordpress
+juju charm-upgrade mysql --channel channel_name
 ```
 
-# Deploying with a configuration file
+
+## Deploying from local charms
```

Comment:
i would do this completely the other way. At the moment, people are redirected to a page which contains a lot of extra stuff about charm tools etc, when all they wanted to do was deploy a charm from a file. It is a particularly short thing too, so it makes more sense to put the relevant part here, and link to it from the other document.